client/dashboard/app/queries/site-logs.ts

```diff
@@ -0,0 +1,51 @@
+import { queryOptions, keepPreviousData } from '@tanstack/react-query';
+import {
+	fetchSiteLogs,
+	SiteLogsAPIResponse,
+	SiteLogsData,
+	SiteLogsParams,
+	PHPLog,
+	ServerLog,
+} from '../../data/site-logs';
+
+export const siteLogsQuery = ( siteId: number, params: SiteLogsParams ) =>
+	queryOptions< SiteLogsAPIResponse, unknown, SiteLogsData >( {
+		queryKey: [
+			'site',
+			siteId,
+			'logs',
+			params.logType,
```

Comment:
I think we can simply make the whole params as the key in the fourth position.

`[ 'site', siteId, 'logs', params ]` 

It's because the keys are ordered hierarchically in the key array, and the params do no really have logical order within themselves.